src/Analysis/pipeline/action_types.py

```diff
@@ -0,0 +1,453 @@
+import abc
+from enum import IntEnum
+from .data_types_tree import BaseDTT, CompositeDTT
+from .code_formater import Formater
+import math
+
+class ActionType(IntEnum):
+    """Action type"""
+    simple = 0
+    complex = 1
+    
+class ActionStateType(IntEnum):
+    """Action type"""
+    created = 0
+    initialized = 1
+    wait = 2
+    process = 3
+    finished = 4   
+
+class Runner():
+    """
+    Data crate for action process runner description 
+    """
+    def __init__(self, type=ActionType.simple):
+        self.name = ""
+        """Runner name for loging"""
+        self.command = ""
+        """Command for popen"""
+        self.params = {}
+        """dictionary names => types of input ports"""
+        self.type =  type   
+
+__action_counter__ = 0
+"""action counter"""
+        
+class BaseActionType(metaclass=abc.ABCMeta):
+    """
+    Abstract class of action type, that define tasks method for
+    tasks classes
+    """
+
+    _name = ""
+    """Display name of action"""
+    _description = ""
+    """Display description of action"""
+
+    def __init__(self, **kwargs):
+        global __action_counter__
+ 
+        __action_counter__ += 1
+        self.id = __action_counter__
+        """unique action number"""
+        self.state = ActionStateType.created
+        """action state"""
+        self.inputs = []
+        """dictionary names => types of input ports"""
+        self.inputs_os = []
+        """
+        number of output slot where is action join (opposite slot)
+        default is first slot
+        """
+        """dictionary names => types of input ports"""
+        self.outputs = []
+        """dictionary names => DTT typse on output ports"""
+        self.variables = {}
+        """dictionary names => types of variables"""
+        self.type = ActionType.simple
+        """action type"""
+        for name, value in kwargs.items():
+            if name == 'Input':
+                self.inputs.append(value)
```

Comment:
Nezavadel bych Inputs pouze Input a pokud je to neni array, tak self.inputs=[ value ].